src/misc.c

```diff
@@ -815,12 +822,15 @@ int _libssh2_get_bignum_bytes(struct string_buf *buf, unsigned char **outbuf)
     buf->dataptr += data_len;
     buf->offset += data_len;
 
-    return bn_len;
+    if (outlen != NULL)
+        *outlen = (size_t)bn_len;
+
+    return 0;
 }
 
 int _libssh2_check_length(struct string_buf *buf, size_t len)
 {
-    return ((int)(buf->dataptr - buf->data) <= (int)(buf->len - len)) ? 1 : 0;
+    return ((ssize_t)(buf->dataptr - buf->data) <= (ssize_t)(buf->len - len)) ? 1 : 0;
```

Comment:
shouldn't this function then also return a `ssize_t` ? Seems odd to just change the typecasts.

Reply:
It returns 1 or 0 regardless of free space, cleaner API IMHO. The casts are to allow larger values (but still signed), in reality these upper limits should never bit hit.

Reply:
It'd be cool with a comment above `_libssh2_check_length()` explaining its use/purpose...